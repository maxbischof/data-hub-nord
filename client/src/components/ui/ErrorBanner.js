import React, { useState } from 'react'
import styled, { css, keyframes } from 'styled-components'
import { ReactComponent as CloseIcon } from '../../icons/close.svg'
import { ReactComponent as ProblemIcon } from '../../icons/problem.svg'
import RootPage from '../pages/RootPage'

export default function ErrorBanner() {
  const [showBanner, setShowBanner] = useState(true)

  return (
    <>
      <BannerContainer isVisible={showBanner}>
        <Banner>
          <StyledProblemIcon />
          <span>Es ist ein Fehler passiert. Bitte versuche es erneut.</span>
          <StyledCloseIcon onClick={() => setShowBanner(false)} />
        </Banner>
      </BannerContainer>
      <RootPage />
    </>
  )
}

const BannerContainer = styled.div`
  position: absolute;
  width: 100%;
  ${(props) => !props.isVisible && CssAnimation}
`

const Keyframes = keyframes`
  from {
    top: 0;
  }

  to {
    top: -150px;
  }
`

const CssAnimation = css`
  animation-duration: 3s;
  animation-name: ${Keyframes};
  animation-fill-mode: forwards;
`

const Banner = styled.div`
  height: 77px;
  background-color: var(--red);
  color: white;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  flex-shrink: 0;

  span {
    margin: 0 30px 0 0;
  }
`

const StyledCloseIcon = styled(CloseIcon)`
  position: absolute;
  top: 10px;
  right: 10px;
`

const StyledProblemIcon = styled(ProblemIcon)`
  margin: 0 20px;
`
