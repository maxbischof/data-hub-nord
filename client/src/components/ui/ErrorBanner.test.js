import React from 'react'
import ErrorBanner from './ErrorBanner'
import renderer from 'react-test-renderer'

it('renders ErrorBanner component', () => {
  const renderedErrorBanner = renderer
    .create(<ErrorBanner text={'Houston we have a problem!'} />)
    .toJSON()
  expect(renderedErrorBanner).toMatchSnapshot()
})
