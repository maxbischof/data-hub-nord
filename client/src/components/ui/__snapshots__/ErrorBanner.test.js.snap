// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ErrorBanner component 1`] = `
<div
  className="sc-bdnylx gpkwlA"
>
  <div
    className="sc-gtssRu fdXflC"
  >
    <svg
      className="sc-hKFyIo eJoWya"
    >
      problem.svg
    </svg>
    <span>
      Houston we have a problem!
    </span>
    <svg
      className="sc-dlnjPT gLdAVN"
      onClick={[Function]}
    >
      close.svg
    </svg>
  </div>
</div>
`;
